src/Features/CSharp/Portable/CodeMapper/CSharpMapCodeService.NodeToMap.cs

```diff
@@ -0,0 +1,423 @@
+                (RecordDeclarationSyntax rcr) => rcr.Identifier.ToString(),
+                (ConstructorDeclarationSyntax constructor) => BuildConstructorIdentifier(constructor),
+                (LocalFunctionStatementSyntax localFunction) => BuildLocalFunctionIdentifier(localFunction),
+                (FieldDeclarationSyntax fieldDeclaration) => BuildFieldDeclarationIdentifier(fieldDeclaration),
+                _ => node.ToString(),   // Fallback to returning the full string of the node as identifier.
+            };
+        }
+
+        private static string BuildFieldDeclarationIdentifier(FieldDeclarationSyntax fieldDeclaration)
+        {
+            var firstDeclarator = fieldDeclaration.Declaration.Variables.FirstOrDefault();
+            if (firstDeclarator is not null)
+            {
+                return firstDeclarator.Identifier.ToString();
+            }
+
+            // Fallback to returning the full string of the node as identifier.
+            return fieldDeclaration.ToString();
+        }
+
+        private static string BuildLocalFunctionIdentifier(LocalFunctionStatementSyntax localFunction)
+        {
+            var identifier = localFunction.Identifier.ToString();
+
+            // when identifier is empty, that means this
+            // local function declaration represents a Constructor
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                return BuildConstructorIdentifier(localFunction);
+            }
+
+            return identifier;
+        }
+
+        /// <summary>
+        /// Returns a composed version of the constructor's identifier.
+        /// The string is composed by the original identifier of the constructor
+        /// and the types of its parameters.
+        /// Example:
+        /// MyClass(string, string, int)
+        /// </summary>
+        /// <param name="ctor">The constructor declaration syntax</param>
+        /// <returns>An identifier string composed by the constructor's identifier and argument types.</returns>
+        private static string BuildConstructorIdentifier(SyntaxNode ctor)
+        {
+            ParameterListSyntax parameters;
+            var identifier = string.Empty;
+            if (ctor is ConstructorDeclarationSyntax cds)
+            {
+                identifier = cds.Identifier.ToString();
+                parameters = cds.ParameterList;
+            }
+            else if (ctor is LocalFunctionStatementSyntax lfs)
+            {
+                identifier = lfs.ReturnType.ToString();
+                parameters = lfs.ParameterList;
+            }
+            else
+            {
+                return ctor.ToString();
+            }
+
+            var parameterTypes = parameters.Parameters
+                .Select(p => p.Type?.ToString())
+                .Where(type => type is not null);
+            if (parameterTypes.Any())
+            {
+                return $"{identifier}({string.Join(", ", parameterTypes)})";
```

Comment:
this is also worrying.  it looks to be building up a syntactic view of a signature.  

why is this not done for methods?